lockbox-ios/Store/DataStore.swift

```diff
@@ -357,12 +378,23 @@ extension DataStore {
     }
 
     private func setInitialLockState() {
-        let lockedValue = self.keychainWrapper.bool(forKey: lockedKey) ?? false
-        if lockedValue {
-            self.storageStateSubject.onNext(.Locked)
-        } else {
-            self.storageStateSubject.onNext(.Unprepared)
-            self.keychainWrapper.set(false, forKey: lockedKey)
+        guard profile.hasSyncableAccount() else {
```

Comment:
nice! I like this much better than relying on the keychain. This might resolve bug #374, actually.